### PR TITLE
Fix issues in launch_pager().

### PR DESCRIPTION
**Pull request by markus-oberhumer**

### CODE REVIEW

**Comment by markus-oberhumer** [comment]
Hi Tavian,

first many thanks for your nice bfs tool!

Since the last 20 years or so I'm using `export LESS="-MM -i"` in work environment - and this does conflict with the default handling of `bfs --help`. Furthermore `LESS= bfs --help` also did not work.

So I've tried to fix that in this pull request - please have a look.

Cheers,
Markus

File: parse.c [hunk 1]
```diff
@@ -2535,46 +2535,26 @@ static CFILE *launch_pager(pid_t *pid, CFILE *cout) {
...
 	}
 
 	close(pipefd[0]);
-	if (envp != environ) {
-		free(envp);
-	}
 	bfs_spawn_destroy(&ctx);
 	return ret;
 
 fail_envp:
```
Comment (tavianator) [inline parse.c:2557]: This label is empty now, might as well remove it

File: spawn.c [hunk 2]
```diff
@@ -125,6 +128,16 @@ int bfs_spawn_addfchdir(struct bfs_spawn *ctx, int fd) {
 	}
 }
 
+int bfs_spawn_addputenv(struct bfs_spawn *ctx, char *s) {
```
Comment (tavianator) [inline spawn.c:131]: I would make this take `const char *` and `strdup()` it

**Comment by tavianator** [comment]
Also, do you mind adding this patch next time you push?  It should keep the FreeBSD build from hanging forever, since it doesn't have access to the necessary secrets when running a fork.  I thought GitHub did this automatically but it just runs it with empty strings.

```diff
diff --git a/.github/workflows/freebsd.yml b/.github/workflows/freebsd.yml
index 945be19..0939303 100644
--- a/.github/workflows/freebsd.yml
+++ b/.github/workflows/freebsd.yml
@@ -4,6 +4,8 @@ on: [push, pull_request]
 
 jobs:
   build:
+    if: github.repository_owner == "tavianator"
+
     runs-on: ubuntu-latest
 
     steps:
```

**Comment by markus-oberhumer** [comment]
I've addressed your review request and pushed a new verions of my branch.  BTW, I could also squash the commits to make the history look nicer if you prefer.

As for git, I never had any issues with its pager handling - but I'll have a look at the git sources.

**Comment by markus-oberhumer** [comment]
BTW, it seems that YAML syntax is not correct - please see https://github.com/tavianator/bfs/actions/runs/745604714 .

> Also, do you mind adding this patch next time you push? It should keep the FreeBSD build from hanging forever, since it doesn't have access to the necessary secrets when running a fork. I thought GitHub did this automatically but it just runs it with empty strings.
> 
> ```diff
> diff --git a/.github/workflows/freebsd.yml b/.github/workflows/freebsd.yml
> index 945be19..0939303 100644
> --- a/.github/workflows/freebsd.yml
> +++ b/.github/workflows/freebsd.yml
> @@ -4,6 +4,8 @@ on: [push, pull_request]
>  
>  jobs:
>    build:
> +    if: github.repository_owner == "tavianator"
> +
>      runs-on: ubuntu-latest
>  
>      steps:
> ```

**Comment by tavianator** [comment]
> I've addressed your review request and pushed a new version of my branch. BTW, I could also squash the commits to make the history look nicer if you prefer.

Thanks!  I'd like the fixes folded into the relevant commits, but don't squash the whole thing.

> As for git, I never had any issues with its pager handling - but I'll have a look at the git sources.

Strange.  `LESS= git log` prints `ESC[33mcommit ...` for me.

> BTW, it seems that YAML syntax is not correct - please see https://github.com/tavianator/bfs/actions/runs/745604714 .

Ah oops, I think it has to be single quotes.  Didn't realize that would make a difference.

**Comment by tavianator** [comment]
Also if you can s/supprt/support/g in 9fe44c83 that'd be good

**Comment by markus-oberhumer** [comment]
Rebased and cleaned up. Please review but don't merge yet - still looking at git.

**Comment by markus-oberhumer** [comment]
BTW, that FreeBSD/tailscale CI test is very slow (running for more than 15 minutes right now).

**Comment by tavianator** [comment]
Thanks!  Looks good.

> BTW, that FreeBSD/tailscale CI test is very slow (running for more than 15 minutes right now).

It's slow because it doesn't have the secrets, so it stalls waiting for me to manually approve the tailscale connection.  I thought the `if:` would stop it from running but I think the condition is always true.  Can you try this instead?

```
if: github.event.pull_request.head.repo.full_name == github.repository
```

**Comment by markus-oberhumer** [comment]
Sigh - you are right about `git`. I've just realized I have a `pager = less -FRX` setting in my ~/.gitconfig ...

And as closely following git is probably a *very* good idea (even if the `LESS= git log` behaviour is questionable) this pull request should probably get closed.

**Comment by markus-oberhumer** [comment]
> if: github.event.pull_request.head.repo.full_name == github.repository

I committed this in my latest (squashed) version of this issue, and yes that seems to work!

**Comment by markus-oberhumer** [comment]
I'm closing this fow now - will resubmit an improved version soon.

**Pull request closed without merging**